bug: error thrown if "tasks.json" does not contain "tasks" key

### Neovim version (nvim -v)

v0.10.3

### Operating system/version

6.12.7-arch1-1

### Describe the bug

```log
[ERROR] Template provider vscode: ...im/rocks/rocks_rtp/lua/overseer/template/vscode/init.lua:344: bad argument #1 to 
'ipairs' (table expected, got nil)                                                                                    
stack traceback:                                                                                                      
        [C]: in function 'ipairs'                                                                                     
        ...im/rocks/rocks_rtp/lua/overseer/template/vscode/init.lua:344: in function <...im/rocks/rocks_rtp/lua/overse
er/template/vscode/init.lua:322>                                                                                      
        [C]: in function 'xpcall'                                                                                     
        ...hare/nvim/rocks/rocks_rtp/lua/overseer/template/init.lua:548: in function 'list'                           
        ...cal/share/nvim/rocks/rocks_rtp/lua/overseer/commands.lua:307: in function 'run_template'                   
        ...cal/share/nvim/rocks/rocks_rtp/lua/overseer/commands.lua:171: in function <...cal/share/nvim/rocks/rocks_rt
p/lua/overseer/commands.lua:151>  
```

### What is the severity of this bug?

minor (annoyance)

### Steps To Reproduce

1. Create an incorrect `.vscode/tasks.json` (without tasks key):
```json
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Python: task",
      "type": "process",
      "cmd": "python",
    }
  ]
}
```
2. Enter `:OverseerRun`

### Expected Behavior

Nothing happens, or a warning is emitted.

### Minimal example file

_No response_

### Minimal init.lua

```Lua
-- DO NOT change the paths and don't remove the colorscheme
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({
    "git",
    "clone",
    "--filter=blob:none",
    "--single-branch",
    "https://github.com/folke/lazy.nvim.git",
    lazypath,
  })
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
  "folke/tokyonight.nvim",
  { "stevearc/dressing.nvim", config = true },
  {
    "stevearc/overseer.nvim",
    config = function()
      require("overseer").setup({
        -- add your overseer config here
      })
    end,
  },
  -- add any other plugins here
}
require("lazy").setup(plugins, {
  root = root .. "/plugins",
})

vim.cmd.colorscheme("tokyonight")
-- add anything else here
```


### Additional context

_No response_

--- a/lua/overseer/template/vscode/init.lua
+++ b/lua/overseer/template/vscode/init.lua
@@ -341,6 +341,13 @@ return {
     end
     local ret = {}
     local precalculated_vars = variables.precalculate_vars()
+
+    if content.tasks == nil then
+      vim.notify("No 'tasks' key found in '.vscode/tasks.json'", vim.log.levels.WARN)
+      cb({})
+      return
+    end
+
     for _, task in ipairs(content.tasks) do
       local defn = vim.tbl_deep_extend("force", global_defaults, task)
       defn = vim.tbl_deep_extend("force", defn, task[os_key] or {})

